types/src/proof/position/mod.rs

```diff
@@ -75,8 +76,13 @@ impl Position {
         self.0
     }
 
-    pub fn from_postorder_index(index: u64) -> Self {
-        Position(postorder_to_inorder(index))
+    pub fn from_postorder_index(index: u64) -> Result<Self> {
+        ensure!(
+            index < (!0u64 >> 2),
```

Comment:
The math still seems wrong....

probably `index < (!0u64 -1)`?

Reply:
Maybe `u64::MAX` is a bit easier to read than `!0u64`...